### PR TITLE
add DTO, service, mapper, repository for category model

### PR DESCRIPTION
**Pull request by VolodymyrDavydov**

### CODE REVIEW

File: src/main/java/com/example/demo/service/category/CategoryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                new EntityNotFoundException("Can't find category with id " + category.getId()));
+        categoryFromDb.setName(category.getName());
+        categoryFromDb.setDescription(category.getDescription());
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+
```
Comment (overpathz) [inline src/main/java/com/example/demo/service/category/CategoryServiceImpl.java:55]:
```suggestion
```

File: src/main/java/com/example/demo/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToMany(mappedBy = "categories")
+    private Set<Book> booksSet;
```
Comment (overpathz) [inline src/main/java/com/example/demo/model/Category.java:28]:
```suggestion
    private Set<Book> booksSet = new HashSet<>();
```

File: src/main/java/com/example/demo/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+package com.example.demo.controller;
+
```
Comment (overpathz) [inline src/main/java/com/example/demo/controller/CategoryController.java:2]: image

**Review by overpathz — CHANGES_REQUESTED** [review]
Need some fixes

File: src/main/java/com/example/demo/model/Category.java (hunk 2 above)
Comment (overpathz) [inline src/main/java/com/example/demo/model/Category.java]:
```suggestion
    private Set<Book> booksSet = new HashSet<>();
```

File: src/main/java/com/example/demo/service/category/CategoryServiceImpl.java (hunk 1 above)
Comment (overpathz) [inline src/main/java/com/example/demo/service/category/CategoryServiceImpl.java]:
```suggestion
```

**Review by overpathz — CHANGES_REQUESTED** [review]
Also, you need to implement soft delete approach

**Review by Ivan95kos — APPROVED** [review]
Great job

**Pull request merged**